integration/embedded_test.go

```diff
@@ -76,3 +79,60 @@ func TestEmbedded(t *testing.T) {
 	cancel()
 	<-done
 }
+
+func TestEmbeddedTLS(t *testing.T) {
+	t.Parallel()
+
+	cert, key := setup.GetTLSFilesPaths()
+
+	f, err := ferretdb.New(&ferretdb.Config{
+		Listener: ferretdb.ListenerConfig{
+			TLS:         "127.0.0.1:47017",
+			TLSCertFile: cert,
+			TLSKeyFile:  key,
+		},
+		Handler:       "pg",
+		PostgreSQLURL: testutil.PostgreSQLURL(t, nil),
+	})
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(testutil.Ctx(t))
+	defer cancel()
+
+	// check that Run exits on context cancel
+	done := make(chan struct{})
+
+	go func() {
+		err = f.Run(ctx)
+		t.Logf("Run exited with %v.", err) // result is undefined for now
+		cancel()
+		close(done)
+	}()
+
+	tlsConfig, err := options.BuildTLSConfig(map[string]interface{}{
+		"tlsCAFile": path.Join("..", "build", "certs", "rootCA.pem"),
+	})
+	require.NoError(t, err)
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(f.MongoDBURI()).SetTLSConfig(tlsConfig))
```

Comment:
I like this example, it helps understand how to connect via TLS :)

I tried running FerretDB locally and connect through mongosh:

To run:
```
  run:
    desc: "Run FerretDB"
    deps: [build-testcover]
    cmds:
      - >
        bin/ferretdb-testcover{{exeExt}} -test.coverprofile=cover.txt -- --test-records-dir=records
        --mode=diff-normal --listen-addr=:27017 --listen-tls=:47017 --listen-tls-cert-file=build/certs/server-cert.pem --listen-tls-key-file=build/certs/server-key.pem
```

To connect:
```
mongosh -port 47017 --tls --tlsCAFile build/certs/rootCA.pem
```

It works!

So, currently, we don't check client's certificate, I guess it's for https://github.com/FerretDB/FerretDB/issues/40, right?

Reply:
Yes, checking client certificates is for the authentication part.

Reply:
Not even for authentication part – it is unlikely we will implement authentication via client certs anytime soon